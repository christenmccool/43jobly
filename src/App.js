import React from 'react';
import NavBar from './NavBar';
import Routes from './Routes';

function App() {
 
  return (
    <div>
      <NavBar />
      <Routes />
    </div>
  );
}

export default App;
