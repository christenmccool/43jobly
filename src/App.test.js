import { render, asFragment } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

test('renders without crashing', () => {
  render(<MemoryRouter><App /></MemoryRouter>);
});

test('matches snapshot', () => {
  const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

test('renders nav bar and home page', () => {
  const {getByText} = render(<MemoryRouter><App /></MemoryRouter>);

  expect(getByText('Jobly')).toBeInTheDocument();
  expect(getByText('Login')).toBeInTheDocument();
  expect(getByText('Signup')).toBeInTheDocument();
  expect(getByText('Home Page Welcome Message')).toBeInTheDocument();
});
