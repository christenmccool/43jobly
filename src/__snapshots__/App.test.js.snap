// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="NavBar"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Jobly
      </a>
      <a
        href="/login"
      >
        Login
      </a>
      <a
        href="/signup"
      >
        Signup
      </a>
      <a
        href="/companies"
      >
        Companies
      </a>
      <a
        href="/jobs"
      >
        Jobs
      </a>
      <a
        href="/profile"
      >
        Profile
      </a>
    </nav>
    <h1>
      Home Page Welcome Message
    </h1>
  </div>
</DocumentFragment>
`;
